example/hooks.ts

```diff
@@ -0,0 +1,158 @@
+import { useCallback, useEffect, useMemo, useRef, useState } from "react";
+
+import {
+  ComputedFeature,
+  LazyLayer,
+  MapRef,
+  SketchEditingFeature,
+  SketchType,
+  ViewerProperty,
+} from "@reearth/core";
+
+import { DEFAULT_CAMERA, DEFAULT_LAYERS, DEFAULT_TILE } from "./constants";
+import { DEFAULT_VIEWER_PROPERTY } from "./scene";
+import { TEST_LAYERS } from "./testLayers";
+import { CESIUM_ION_ACCESS_TOKEN } from "./token";
+
+export default () => {
+  const ref = useRef<MapRef>(null);
+  const [isReady, setIsReady] = useState(false);
+  const handleMount = useCallback(() => {
+    requestAnimationFrame(() => {
+      setIsReady(true);
+    });
+  }, []);
+
+  const handleAPIReady = useCallback(() => {
+    ref.current?.sketch.overrideOptions({
+      dataOnly: false,
```

Comment:
What does `dataOnly` mean?

Reply:
We can config the bahavior after user finished draw something:
- if `dataOnly` is true, core will not add a layer to map
- if `dataOnly` is false, core will add a layer to map
In both cases user can get the new feature's data through `handleSketchFeatureCreate`

For example:
- Re:Earth Visualizer sets this as true, it will get the data and manage the layer by itself.
- View 3.0 sets this as true, it will manage the layer by itself as well.
- Core example project: it doesn't want to manage layer so it can ask core to add sketch layers after drawing.